core/src/main/java/org/apache/brooklyn/core/objs/BasicSpecParameter.java

```diff
@@ -384,5 +387,34 @@ public static WeightedParameter getFieldConfig(ConfigKey<?> config, Field config
         }
     }
 
+    /**
+     * Adds the given list of {@link SpecParameter parameters} to the provided
+     * {@link AbstractBrooklynObjectSpec spec}. Inherits parent parameters except
+     * where there is an existing {@link ConfigKey config key}.
+     * <p>
+     * Implemented by explicitly replacing the existing parameters with an
+     * updated list.
+     *
+     * @see EntitySpec#parameters(List)
+     */
+    public static void addParameters(AbstractBrooklynObjectSpec<?, ?> spec, List<? extends SpecParameter<?>> explicitParams, BrooklynClassLoadingContext loader) {
+        if (explicitParams.size() > 0) {
+            List<SpecParameter<?>> currentParams = MutableList.copyOf(spec.getParameters());
+            Map<ConfigKey<?>, ?> currentConfig = spec.getConfig();
+            for (final SpecParameter<?> param : explicitParams) {
+                Optional<ConfigKey<?>> configExists = Iterables.tryFind(currentConfig.keySet(), ConfigPredicates.nameEqualTo(param.getConfigKey().getName()));
```

Comment:
Should add parameters even if a config key with the same name exists. Config keys coming from yaml are too generic if the config key doesn't exist on the class. An explicit parameter with type information is always preferred. And even if a config key with the same name exists on the class, a parameter from yaml should override it.